static/js/publisher/form/media.js

```diff
@@ -13,15 +13,14 @@ class Media extends React.Component {
     this.mediaChange = this.mediaChange.bind(this);
     this.keyboardEvent = this.keyboardEvent.bind(this);
 
+    this.toggleRestrictions = this.toggleRestrictions.bind(this);
+
     this.state = {
-      mediaData: props.mediaData
+      mediaData: props.mediaData,
+      restrictionsVisible: false
     };
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    this.props.updateState(prevState.mediaData);
```

Comment:
Why can't it be called on each component update?
I understand that state contains more than just `mediaData`, so there will be some unnecessary calls (when opening/closing image restrictions), but is this such a heavy call that we need to avoid it?
Also, if mediaData didn't change, maybe it should be up to `updateState` function to decide whether to do something or not?

Reply:
As it's passed through to the form to update the state and update the form, which includes diffing and updating localStorage etc. I figured we're better off only updating and causing that chain reaction once, when we know we need to. And so we don't need to add some extra logic to do what the `updateState` callback does for free.

Reply:
OK, still I feel like it should be `updateState` to check if anything changed and only update if necessary.
But if we want to keep that in the component maybe it would make sense to keep it in one place (`componentDidUpdate`) and do something like:
```
if (prevState.mediaData !== this.state.mediaData) {
    this.props.updateState(this.state.mediaData);
}
```

Of course that assumes that mediaData is easily comparable, so needs to be updated in immutable way.

Reply:
I'll do a little test, my worry is that we're passing `File`'s through the state. I'm not sure how comparing them would work

Reply:
@Lukewh It's just comparing 2 arrays, so items aren't even looked at. Only if array (reference) changed they will be considered different. If we modify array by pushing or changing items in place this comparison would not work.

Way around it may be serialising with `JSON.stringify` and comparing that, but that's some overhead and not fully bulletproof, so if it get's that complicated let's leave it as it is now.